assets/src/blocks/Media/MediaEditor.js

```diff
@@ -0,0 +1,155 @@
+          <RichText
+            tagName="h2"
+            className="page-section-header"
+            placeholder={__('Enter title', 'planet4-blocks-backend')}
+            value={video_title}
+            onChange={toAttribute('video_title')}
+            keepPlaceholderOnFocus={true}
+            withoutInteractiveFormatting
+            characterLimit={40}
+            multiline="false"
+          />
+        </header>
+      </Tooltip>
+      <RichText
+        tagName="p"
+        className="page-section-description"
+        placeholder={__('Enter description', 'planet4-blocks-backend')}
+        value={description}
+        onChange={toAttribute('description')}
+        keepPlaceholderOnFocus={true}
+        withoutInteractiveFormatting
+        characterLimit={200}
+      />
+      <MediaPlaceholder
+        labels={{ title: __('Video poster image [Optional]', 'p4ge'), instructions: __('Applicable for .mp4 video URLs only.', 'p4ge')}}
+        icon="format-image"
+        onSelect={ onSelectImage }
+        onSelectURL={ onSelectURL }
+        onError={ onUploadError }
+        accept="image/*"
+        allowedTypes={["image"]}
+      />
+      {
+        (youtube_id && !youtube_id.endsWith('.mp4')) && !embedHtml
+        ? <div className="block-edit-mode-warning components-notice is-error">
+            { __( 'The video URL could not be parsed.', 'planet4-blocks' ) }
+          </div>
+        : <Fragment>
+            {
+              youtube_id && youtube_id.endsWith('.mp4')
+              ? <MediaElementVideo videoURL={ youtube_id } videoPoster={ posterURL } />
+              : <MediaEmbedPreview html={ embedHtml ? embedHtml : null } />
+            }
+          </Fragment>
+      }
+    </Fragment>
+  )
+}
+
+export const MediaEditor = (props) => {
+  const { isSelected, attributes, setAttributes } = props;
+  const { youtube_id, video_poster_img } = attributes;
+  let mediaURL = youtube_id;
+
+  const toAttribute = attributeName => value => setAttributes({ [attributeName]: value });
+
+  // Assume that a non-URL is a YouTube video ID, for back compat.
+  if ( youtube_id && youtube_id.indexOf('/') === -1 ) {
+    mediaURL = `https://www.youtube.com/watch?v=${mediaURL}`;
+  }
+
+  if ( youtube_id && youtube_id.indexOf('youtube.com') > -1 ) {
+    mediaURL.replace('youtube.com', 'youtube-nocookie.com');
+  }
+
+  const { embedHtml, posterURL } = useSelect((select) => {
+    const embedPreview = select('core').getEmbedPreview( mediaURL );
+    const media = select('core').getMedia(video_poster_img)
+    return {
+      embedHtml: embedPreview ? embedPreview.html : null,
+      posterURL: media ? media.media_details.sizes.large.source_url : null
+    }
+  },
+  [ youtube_id, video_poster_img ]);
+
+  useEffect(() => {
+    setAttributes({
+      embedHtml: embedHtml ? embedHtml : null,
+      posterURL: posterURL ? posterURL : null
+    })
+  },
+  [ embedHtml, posterURL ]);
+
+  return (
+    <div>
+      {
+        renderEdit(props)
```

Comment:
Shouldn't this be rendered only when `isSelected` is true? 🤔

Reply:
So here's a thing which I actually asked in the React conference on Friday (I didn't get a proper answer): if we use `isSelected` here, React complains about the order of hooks changing (I hate that error!). I see `renderEdit` as a function component even if we are calling it as a function... I thought the "hooks order" error wouldn't trigger in a child component if it's rendered conditionally, but maybe it is triggered. 

So I moved the `isSelected` check to inside `renderEdit` after the hooks are set, right before the `return` call.

Reply:
Ok, I found another solution: instead of calling `renderEdit`, I turned it into a component by using it as a JSX tag. It's the same function code but invoked with a different syntax. This fixes the "order of hooks" warning. I guess it changes depending on the way React wraps components when it founds a JSX tag instead of you invoking a function directly.